Rando: Adds 90~ new random rupee names, alphabetizes them

nianjiilical opened the pull request:
As the title says, massively expands the list of possible rupee names when Randomize Rupee Names is enabled. It's a fun and silly feature, so why not expand it? Also alphabetized the list because I was writing a lot anyway.

Most of the added names were taken by skimming a few currency or video game-related TVTropes pages for inspiration, picking ones that either come from popular works, sound funny, or both, plus a few little jokes.

First time contributing to the repo so I hope changing the text formatting slightly in the file doesn't ruin everything somehow, plus a few of them have umlats and characters that I *think* OoT supports but may be wrong on.

The list of new names added in case the people in charge want to approve them or just know the references:

<details>
  <summary>new names</summary>

- Bad RNG Rolls (joke)
- Beanbean Coins (Mario & Luigi)
- Beans (several things)
- Beli (One Piece)
- Berries (Bug Fables)
- Bison Dollars (Street Fighter)
- Blue Essence (League of Legends)
- Bolts (Ratcher & Clank)
- Boondollars (Homestuck)
- BugFrags (Mega Man Battle Network)
- Bǎn (Avatar The Last Airbender)
- Chaos Orbs (Path of Exile)
- Clams (a lot of things)
- Cookies (Cookie Clicker)
- Cor (Sword Art Online)
- Crimebucks (Carmen Sandiego)
- Crystal Shards (Kirby 64)
- Cubits (Battlestar Galactica)
- DNA (Spore)
- Dalmations (Kingdom Hearts 2/101 Dalmations)
- Dampecoin (ZFG's streams)
- Dark Elixer (Clash of Clans)
- Darseks (Star Trek)
- Dead Memes (joke)
- Dosh (Killing Floor)
- Dwarfbucks (Dwarf Fortress)
- Energon (Transformers)
- Eris (Konosuba)
- Eurodollars (Cyberpunk)
- Extinction Points (Dino Crisis)
- Flurbos (Rick and Morty)
- Frog Coins (Super Mario RPG)
- Gekz (Halo)
- Gold Dragons (A Song of Ice and Fire)
- Goober Dollars (Spongebob Squarepants)
- Glitches (joke)
- Green Herbs (Resident Evil)
- Hell (Disgaea)
- Hylian Loaches (OoT)
- ISK (EVE Online)
- Ice Traps (joke)
- Kinstones (Zelda: Minish Cap)
- Kremcoins (Donkey Kong Country 2)
- Leaves (Octopath Traveller)
- Lien (RWBY)
- Lungmen Dollars (Arknights)
- Macca (Shin Megami Tensei)
- Mann Co. Keys (Team Fortress 2)
- Meat (Kingdom of Loathing)
- Meat Stacks (Kingdom of Loathing)
- Medaparts (Medabots)
- Monopoly Money (Monopoly)
- Mora (Genshin Impact)
- Mysteries (joke)
- Neopoints (Neopets)
- Nuyen (Shadowrun)
- Pix (Recettear)
- Pixels (Starbound)
- Power Pellets (Pac-Man)
- Primogems (Genshin Impact)
- Refined Metal (Team Fortress 2)
- Retweets (Twitter)
- Rhinu (Discworld)
- Riot Points (League of Legends)
- Runite Ore (Runescape)
- Ryō (Naruto)
- Réals (Disco Elysium)
- Saint Quartz (Fate/Grand Order)
- Septims (Elder Scrolls)
- Spent Casings (Enter the Gungeon)
- Spice (Dune)
- Spondulicks (Problem Sleuth)
- Star Bits (Super Mario Galaxy)
- Star Chips (Yugioh)
- Stones of Jordan (Diablo 2)
- Super Sea Snails (Splatoon)
- Teef (Warhammer 40k)
- Telecrystals (Space Station 13)
- Tiberium (Command & Conquer)
- TokKul (Runescape)
- Turnips (Animal Crossing)
- Upvotes (Reddit)
- Watts (Pokemon Sword/Shield)
- Widgets (Bionicle)
- Woolongs (Cowboy Bebop)
- World Dollars (Guilty Gear)
- Wumpa Fruit (Crash Bandicoot)
- [P] (Touhou)

</details>

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh-linux-compatibility.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/521410185.zip)
  - [soh-linux-performance.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/521410186.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/521410187.zip)
  - [soh-switch.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/521410188.zip)
  - [soh-wiiu.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/521410189.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/521410190.zip)
<!--- section:artifacts:end -->

nianjiilical committed:
more rupee names

nianjiilical committed:
typo

stratomaster64 commented on soh/soh/Enhancements/randomizer/randomizer.cpp:
```diff
@@ -47,19 +47,40 @@ const std::string Randomizer::rupeeMessageTableID = "RandomizerRupees";
-    "Ether",        "Doge",          "Cards",        "Talent",     "Poko",         "Lira",        "Kroner",
-    "Store Credit", "Social Credit", "Cocoa Beans",  "Strawbs"
+static const char* englishRupeeNames[165] = { 
+    "?????",                "Bad RNG Rolls",        "Bananas",              "Beanbean Coins",       "Beans",
```
`?????` should be taken out, more confusing than funny

stratomaster64 commented on soh/soh/Enhancements/randomizer/randomizer.cpp:
```diff
@@ -47,19 +47,40 @@ const std::string Randomizer::rupeeMessageTableID = "RandomizerRupees";
-    "Store Credit", "Social Credit", "Cocoa Beans",  "Strawbs"
+static const char* englishRupeeNames[165] = { 
+    "?????",                "Bad RNG Rolls",        "Bananas",              "Beanbean Coins",       "Beans",
+    "Bells",                "Belly",                "Berries",              "Bison Dollars",        "Bitcoin",
```
`Belly` could be `Beli` to more accurately refer to One Piece

stratomaster64 replied:
`Bison Dollars` are specifically from the '94 SF Movie and isn't as well known, not sure about this one

stratomaster64 commented on soh/soh/Enhancements/randomizer/randomizer.cpp:
```diff
@@ -47,19 +47,40 @@ const std::string Randomizer::rupeeMessageTableID = "RandomizerRupees";
+    "Macca",                "Mana",                 "Mann Co. Keys",        "Meat",                 "Meat Stacks",
+    "Meseta",               "Mesetas",              "Minerals",             "Monopoly Money",       "Moons",
+    "Mora",                 "Mumbo Tokens",         "Munny",                "Mushrooms",            "Neopoints",
+    "Noted Runite Ore",     "Notes",                "Nuyen",                "Orbs",                 "Pix",
```
`Noted Runite Ore` is super specific, could do either of `Runite Ore/Rune Bars/Runite Certs`

stratomaster64 commented on soh/soh/Enhancements/randomizer/randomizer.cpp:
```diff
@@ -47,19 +47,40 @@ const std::string Randomizer::rupeeMessageTableID = "RandomizerRupees";
+    "Super Sea Snails",     "Talent",               "Teef",                 "Telecrystals",         "Tiberium",
+    "TokKul",               "Toys",                 "Turnips",              "Upvotes",              "V-Bucks",
+    "Vespene Gas",          "Watts",                "Widgets",              "Woolongs",             "World Dollars",
+    "Wumpa Fruit",          "Zenny",                "Zorkmids",             "[404 ERROR]",          "[P]"
```
`[404 ERROR]` should be taken out, doesn't really make sense to me imo

stratomaster64 reviewed (commented):
So far so good! Just a few things regarding additions.

nianjiilical committed:
feedback

nianjiilical commented:
Thanks for the feedback, I went ahead and made most of those changes. Replaced ????? and [404 ERROR] with Mysteries and Glitches to keep the theme, and renamed Runite Ore and Beli.

The only one I disagree with is Bison Dollars; it's obscure, but from what I've heard the movie is a cult classic and it's a pretty memorable part, to where it's gotten nods here and there officially. I think it's okay to have one or two more obscure picks in there, so hopefully that's fine.

aMannus commented on soh/soh/Enhancements/randomizer/randomizer.cpp:
```diff
@@ -47,19 +47,40 @@ const std::string Randomizer::rupeeMessageTableID = "RandomizerRupees";
+    "Bad RNG Rolls",        "Bananas",              "Beanbean Coins",       "Beans",                "Bells",
+    "Beli",                 "Berries",              "Bison Dollars",        "Bitcoin",              "Blue Essence",
+    "Bolts",                "Bones",                "Boondollars",          "Bottle Caps",          "Bratwürste",
+    "Bucks",                "BugFrags",             "Bullets",              "Bǎn",                  "Cards",
```
I think it might be better to exclude Bullets just to make sure we keep it family friendly. KF7 Ammo already treaded that line a little in my eyes.

aMannus commented on soh/soh/Enhancements/randomizer/randomizer.cpp:
```diff
@@ -47,19 +47,40 @@ const std::string Randomizer::rupeeMessageTableID = "RandomizerRupees";
+    "Doge",                 "Dogecoin",             "Doll Hairs",           "Dollars",              "Dollerydoos",
+    "Dosh",                 "Doubloons",            "Dwarfbucks",           "Emeralds",             "Energon",
+    "Eris",                 "Ether",                "Eurodollars",          "Experience",           "Extinction Points",
+    "Flanian Pobble Beans", "Floopies",             "Flurbos",              "Friends",              "Frog Coins",
```
Just making sure, did you test if the longer names like `Flanian Pobble Beans` fit in the textbox ingame?

aMannus reviewed (changes requested):
Overall this is looking good! Always fun to have more names.

I've left 2 comments of my own.

Regarding `Bison Dollars`, I think it's fine to have some more obscure ones. Hell, `Floopies` is an imaginary currency me and a friend of mine used when we were little, lol. Not everyone is gonna get every single currency name, and that's fine in my eyes.

nianjiilical committed:
more feedback

nianjiilical commented:
Unfortunately because I am very bad at code I haven't successfully managed to build SoH in the last hour or two, so I just cut Flobble Beans and Stones of Baranziah to be safe. Stones of Jordan and Extinction Points are at most one or two characters short of Canadian Dollars so they should fit fine?

Also cut Bullets re: above suggestion and added Kremcoins (Donkey Kong Country 2) and Medaparts (Medabots) because thats what I've been playing recently just to keep the number nice and round (also I removed Canadian Dollars by mistake apparently? whoops)
